app/models/plan.rb

```diff
@@ -486,11 +486,26 @@ def authors
   # The number of answered questions from the entire plan
   #
   # Returns Integer
-  def num_answered_questions
-    Answer.where(id: answers.map(&:id))
-          .includes(:question_options, question: :question_format)
-          .to_a
-          .sum { |answer| answer.is_valid? ? 1 : 0 }
+  def num_answered_questions(phase = nil)
+    if phase.present?
+      # If a phase was specified then return the number of
+      # answered questions within the context of the phase
+      question_ids = phase.sections.map do |s|
+        s.questions.pluck('questions.id')
```

Comment:
I think that the query would be more efficient if we defined the relationship between Phase and questions, or phase and answers, probably both.